examples/bridge-app/linux/UserInputBackend.cpp

```diff
@@ -0,0 +1,328 @@
+#include "Backend.h"
+#include "DynamicDevice.h"
+
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace {
+
+std::vector<std::unique_ptr<Device>> g_devices;
+std::vector<std::unique_ptr<DynamicDeviceImpl>> g_device_impls;
+std::unique_ptr<DynamicDeviceImpl> g_pending;
+
+PropagateWriteCB g_write_cb = [](CommonCluster * cluster, chip::EndpointId ep, chip::ClusterId clusterId, chip::AttributeId attr,
+                                 const uint8_t * buffer) -> EmberAfStatus {
+    printf("Write to ep %d cluster %d attribute %d\n", ep, clusterId, attr);
+    cluster->WriteFromBridge(attr, buffer);
+    return EMBER_ZCL_STATUS_SUCCESS;
+};
+
+void NewDevice(const std::vector<std::string> & tokens)
+{
+    if (g_pending)
+    {
+        printf("Already pending device!\n");
+        return;
+    }
+    if (tokens.size() < 2)
+    {
+        g_pending = std::make_unique<DynamicDeviceImpl>();
+    }
+    else if (tokens[1] == "switch")
+    {
+        g_pending = std::make_unique<DynamicSwitchDevice>();
+    }
+    else
+    {
+        printf("Unknown device type %s\n", tokens[1].c_str());
+        return;
+    }
+}
+
+void AddCluster(const std::vector<std::string> & tokens)
+{
+    if (!g_pending)
+    {
+        printf("Expected pending device!\n");
+        return;
+    }
+    if (tokens.size() < 2)
+    {
+        printf("Expected id of a device type\n");
+        return;
+    }
+
+    for (const auto & cluster : clusters::kKnownClusters)
+    {
+        if (tokens[1] == cluster.name)
+        {
+            std::unique_ptr<CommonCluster> obj(cluster.ctor(::operator new(cluster.size)));
+            DynamicAttributeList dynamic_attrs;
+            auto attrs = obj->GetAllAttributes();
+            for (auto & attr : attrs)
+                dynamic_attrs.Add(attr);
+            g_pending->AddCluster(std::move(obj), dynamic_attrs, nullptr, nullptr);
+        }
+    }
+}
+
+void CancelDevice(const std::vector<std::string> & tokens)
```

Comment:
Should any validity check be done against tokens (e.g. it should be empty?). Here and for other places where arguments are not expected.